src/Microsoft.Security.Utilities.Core/PreciselyClassifiedSecurityKeys/SEC101_031_NuGetApiKey.cs

```diff
@@ -36,6 +36,22 @@ public override Tuple<string, string> GetMatchIdAndName(string match)
             return base.GetMatchIdAndName(match);
         }
 
+#if HIGH_PERFORMANCE_CODEGEN
```

Comment:
>HIGH_PERFORMANCE_CODEGEN [](http://example.com/codeflow?start=4&length=24)

OK, so here's the deal on this one, these secrets actually accept mixed case. NuGet somewhere normalizes keys on receipt before attempting validation. I just confirmed this with a test key, you can change absolutely any key case and things continue to work.

that means you can converge on a single pattern which simply accepts `[a-z2-7]`

alternately you can use my radically over-engineered regex here it is:

> ^.{3}[a-pA-P][0-9a-zA-Z]{15}[aqAQ][0-9a-zA-Z]{11}[euEU][bdfhjlnprtvxz357BDFHJLNPRTVXZ][a-pA-P][0-9A-Za-z]{11}[aeimquy4AEIMQUY]

Reply:
the other issue with accepting my regex is simply that you hate it.